File: 2598-shortest-distance-to-target-string-in-a-circular-array/shortest-distance-to-target-string-in-a-circular-array.ts
function closestTarget(words: string[], target: string, startIndex: number): number {
    const n:number = words.length
    let ans:number = n
    for (let i=0;i<n;i++){
        if(words[i]==target){
            let dis = Math.min(Math.abs(i-startIndex), n - (Math.abs(i-startIndex)))
            ans = Math.min(ans,dis)
        }
    }
    return ans === n ? -1 : ans
};

File: 2598-shortest-distance-to-target-string-in-a-circular-array/shortest-distance-to-target-string-in-a-circular-array.py
class Solution:
    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        n = len(words)
        ans = n
        for i in range(n):
            if words[i] == target:
                dis = min(abs(i-startIndex), n-abs(i-startIndex))
                ans=min(ans,dis)
        if ans == n:
            return -1
        return ans


